lib/tasks/scripts/telemedicine_reports.rb

```
@@ -1,27 +1,56 @@
 # frozen_string_literal: true
 
 require "csv"
+require "net/http"
 
 class TelemedicineReports
-  attr_reader :mixpanel_csv_path, :period_start, :period_end, :mixpanel_data, :report_array
+  attr_reader :period_start, :period_end, :mixpanel_data, :report_array
 
-  def initialize(mixpanel_csv_path, period_start, period_end)
-    @mixpanel_csv_path = mixpanel_csv_path
+  def initialize(period_start, period_end)
     @period_start = period_start
     @period_end = period_end
     @mixpanel_data = {hydrated: [], formatted: []}
     @report_array = []
+    @query = <<~QUERY
+      function main() {
+        return Events({
+          from_date: "#{period_start.strftime("%Y-%m-%d")}",
+          to_date: "#{period_end.strftime("%Y-%m-%d")}"
+        }).filter(function(event) {
+          return event.name == "UserInteraction" && event.properties.name == "Patient Summary:Contact Doctor Clicked"
+        }).groupBy(
+          ["properties.$user_id",
+          function(event) {
+              return (new Date(event.time)).toDateString();
+            }],
+          mixpanel.reducer.count()
+        );
+      }
+    QUERY
+  end
+
+  def fetch_mixpanel_data
+    url = URI.parse("https://mixpanel.com/api/2.0/jql")
+
+    response = begin
+      HTTP.basic_auth(user: ENV["MIXPANEL_API_SECRET"], pass: nil)
+        .post(url, json: {format: "csv", script: @query})
+    rescue HTTP::Error => e
```

Don't need to `rescue` and `raise` it back here, just let it bubble up.